From ba5a0d22464699900a6f21962aa7f3bcef853c74 Mon Sep 17 00:00:00 2001
From: Rob Evers <revers@redhat.com>
Date: Wed, 20 Aug 2014 18:08:52 -0400
Subject: [scsi] be2iscsi: Fix the sparse warning introduced in previous submission

Message-id: <1408558136-21958-1-git-send-email-revers@redhat.com>
Patchwork-id: 88414
O-Subject: [RHEL7.1 e-stor PATCH 1/5] scsi: be2iscsi: Fix the sparse warning introduced in previous submission
Bugzilla: 1130072
RH-Acked-by: Ewan Milne <emilne@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: John Soni Jose <sony.john-n@emulex.com>

http://marc.info/?l=linux-scsi&m=140747401805704&w=2

tested at Emulex

Where sparse warning was introduced:

 commit 73af08e11c6638e2abd6b1fa13cdab58c2bbdbf8
 Author: Jayamohan Kallickal <jayamohan.kallickal@emulex.com>
 Date:   Mon May 5 21:41:26 2014 -0400

  be2iscsi: Fix interrupt Coalescing mechanism.

Signed-off-by: John Soni Jose <sony.john-n@emulex.com>
Signed-off-by: Jayamohan Kallickal <jayamohan.kallickal@emulex.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/be2iscsi/be_cmds.h b/drivers/scsi/be2iscsi/be_cmds.h
index cc7405c..4e8cb61 100644
--- a/drivers/scsi/be2iscsi/be_cmds.h
+++ b/drivers/scsi/be2iscsi/be_cmds.h
@@ -26,9 +26,9 @@
  * The commands are serviced by the ARM processor in the OneConnect's MPU.
  */
 struct be_sge {
- u32 pa_lo;
- u32 pa_hi;
- u32 len;
+ __le32 pa_lo;
+ __le32 pa_hi;
+ __le32 len;
 };
 
 #define MCC_WRB_SGE_CNT_SHIFT 3 /* bits 3 - 7 of dword 0 */
@@ -624,11 +624,11 @@ static inline struct be_sge *nonembedded_sgl(struct be_mcc_wrb *wrb)
 /******************** Modify EQ Delay *******************/
 struct be_cmd_req_modify_eq_delay {
  struct be_cmd_req_hdr hdr;
- u32 num_eq;
+ __le32 num_eq;
  struct {
-  u32 eq_id;
-  u32 phase;
-  u32 delay_multiplier;
+  __le32 eq_id;
+  __le32 phase;
+  __le32 delay_multiplier;
  } delay[MAX_CPUS];
 } __packed;
 
-- 
1.7.1